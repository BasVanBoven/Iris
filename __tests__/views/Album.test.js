import React from 'react';
import { render } from '../test-wrapper';
import Album from '../../src/js/views/Album';
import { state as mockState } from '../state';

jest.mock('react-dnd', () => ({
  ...jest.requireActual('react-dnd'),
  useDrag: jest.fn(),
  useDrop: jest.fn(),
}));
jest.mock('redux-persist', () => ({
  ...jest.requireActual('redux-persist'),
  persistReducer: jest.fn().mockImplementation((config, reducers) => reducers),
}));
jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockImplementation(func => func(mockState)),
  useDispatch: jest.fn(),
  connect: jest.fn(fn => fn()),
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    location: {
      pathname: 'iris.local:6680/iris/album/local:album:md5:66fbea3593ba96a15a9d4188bebab50b',
    },
  }),
  useLocation: () => ({
    pathname: 'iris.local:6680/iris/album/local:album:md5:66fbea3593ba96a15a9d4188bebab50b',
  }),
}));

describe('<Album />', () => {

  const album = mockState.core.items['local:album:md5:66fbea3593ba96a15a9d4188bebab50b'];

  // Need to rebuild Album to functional component, at which point I'll copy previous
  // snapshot testing approach from other project
  it('should render accurately', () => {
    const result = render(<Album album={album} />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
