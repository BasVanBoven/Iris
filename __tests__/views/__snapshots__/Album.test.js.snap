// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Album /> should render accurately 1`] = `
<div
  className="error-message error-message--not-found"
  data-qa-file="ErrorMessage"
  data-qa-node="div"
>
  <i
    className="error-message__icon icon icon--material"
    data-qa-file="ErrorMessage"
    data-qa-node="i"
  >
    error
  </i>
  <h4
    className="error-message__title"
    data-qa-file="ErrorMessage"
    data-qa-node="h4"
  >
    Not found
  </h4>
  <div
    className="error-message__content"
    data-qa-file="ErrorMessage"
    data-qa-node="div"
  >
    <p
      data-qa-file="Album"
      data-qa-node="p"
    >
      Could not find anything with URI 
    </p>
  </div>
</div>
`;
